### DISCUSSION TITLE
General setup question

### DISCUSSION THREAD

**Discussion by bristol1987** [pfelk/pfelk#281] [Q&A] [answered]
Hi! This will be a noob question, but i wanted to ask - what is the setup this project would require? Are both pfsense and ubuntu installed on virtual machines on the same lan? can you provide some information in that regard, like the assumed setup this would work in?

**Comment by a3ilson** [comment, marked as answer]
@bristol1987 -

The setup only requires logs be sent to pfELK. There is no requirement to be on the same network. Let me know if this suffices your answer.

**Reply by neolabrr** [reply]
I guess the question is - how do you set this up in a virtualized environment? fw needs to be on its own vm, ubuntu as well.

**Reply by a3ilson** [reply]
The setup is the same as a physical environment.

Physical:
- Firewall
- Ubuntu w/pfelk
- Host(s)
- Servers 
- Etc...

Virtual
- Virtualized Firewall
- Virtualized Ubuntu w/pfelk
- Host(s)
- Servers
- Etc...

NOTE: Ubuntu w/pfelk was illustrated as an example...pfelk could be installed on Debian, w/LXC, Docker etc...

Regardless of the infrastructure (physical/virtual) a network is still present. That network might be connected through physical devices or virtually. Feel free to elaborate on the question but I do not understand what exactly is being asked.
